src/mavsdk/core/mavlink_parameter_receiver.h

```diff
@@ -23,38 +36,50 @@ class MavlinkParameterReceiver {
     ~MavlinkParameterReceiver();
 
     enum class Result {
-        Success,
-        Timeout,
-        ConnectionError,
-        WrongType,
-        ParamNameTooLong,
-        NotFound,
-        Failed,
-        ParamValueTooLong,
-        UnknownError
+        Success, // All Ok
+        WrongType, // Wrong type provided
+        ParamNameTooLong, // param name provided too long
+        NotFound, // get_xxx param not found
+        ParamValueTooLong // value for param of type string doesn't fit into extended protocol.
     };
 
+    /**
+     * Add a new parameter to the parameter set.
+     * It is recommended to not change the parameter set after first communicating with any client.
+     * (https://mavlink.io/en/services/parameter_ext.html#parameters_invariant).
+     * @param name the unique id for this parameter
+     * @param param_value the value for this parameter
+     * @return Result::ParamNameTooLong if the parameter name is too long,
+     * Result::WrongType if the same parameter name is provided with a different type (aka updating the parameter would mutate the type
+     * of an already provided parameter), Result::ParamValueTooLong if the parameter type is std::string but the value is longer than
+     * the extended protocol allows and
+     * Result::Success otherwise.
+     */
```

Comment:
While this is nice I wouldn't bother too much with docstrings for internal methods.